crates/ere-sp1/src/compile.rs

```diff
@@ -1,102 +1,53 @@
-use std::{fs, path::Path, process::Command};
+use std::{path::Path, process::Command};
 
+use build_utils::docker;
 use tempfile::TempDir;
-use toml::Value as TomlValue;
 use tracing::info;
 
 use crate::error::CompileError;
 
-/// Compile the guest crate and return raw ELF bytes.
-pub fn compile_sp1_program(program_crate_path: &Path) -> Result<Vec<u8>, CompileError> {
-    info!("Compiling SP1 program at {}", program_crate_path.display());
-
-    if !program_crate_path.exists() || !program_crate_path.is_dir() {
-        return Err(CompileError::InvalidProgramPath(
-            program_crate_path.to_path_buf(),
-        ));
-    }
-
-    let guest_manifest_path = program_crate_path.join("Cargo.toml");
-    if !guest_manifest_path.exists() {
-        return Err(CompileError::CargoTomlMissing {
-            program_dir: program_crate_path.to_path_buf(),
-            manifest_path: guest_manifest_path.clone(),
-        });
-    }
-
-    // ── read + parse Cargo.toml ───────────────────────────────────────────
-    let manifest_content =
-        fs::read_to_string(&guest_manifest_path).map_err(|e| CompileError::ReadFile {
-            path: guest_manifest_path.clone(),
-            source: e,
-        })?;
-
-    let manifest_toml: TomlValue =
-        manifest_content
-            .parse::<TomlValue>()
-            .map_err(|e| CompileError::ParseCargoToml {
-                path: guest_manifest_path.clone(),
-                source: e,
-            })?;
-
-    let program_name = manifest_toml
-        .get("package")
-        .and_then(|p| p.get("name"))
-        .and_then(|n| n.as_str())
-        .ok_or_else(|| CompileError::MissingPackageName {
-            path: guest_manifest_path.clone(),
-        })?;
-
-    info!("Parsed program name: {program_name}");
-
-    // ── build into a temp dir ─────────────────────────────────────────────
-    let temp_output_dir = TempDir::new_in(program_crate_path)?;
-    let temp_output_dir_path = temp_output_dir.path();
-    let elf_name = format!("{program_name}.elf");
-
-    info!(
-        "Running `cargo prove build` → dir: {}, ELF: {}",
-        temp_output_dir_path.display(),
-        elf_name
-    );
-
-    let status = Command::new("cargo")
-        .current_dir(program_crate_path)
+pub fn compile(guest_program_full_path: &Path) -> Result<Vec<u8>, CompileError> {
+    // Build the SP1 docker image
+    let tag = "ere-build-sp1:latest";
+    docker::build_image("docker/sp1/Dockerfile", tag)
+        .map_err(|e| CompileError::DockerImageBuildFailed(Box::new(e)))?;
```

Comment:
This is to build the base and corresponding zkVM containers. It is a call that we'll do in all zkVMs. That comes below is zkVM specific ways to compile the guest program.

Reply:
I'll add it in this PR :+1:

Reply:
Done.